File: app-sdk/src/main/java/code/sdk/bridge/BridgeInterface.java
package code.sdk.bridge;

public interface BridgeInterface {

    public void nativeLog(String tag, String msg);

    public void copyText(String text);

    public String getCopiedText();

    public void showNativeToast(String toast);

    public void initAdjustID(String adjustAppID);

    public void trackAdjustEvent(String eventToken, String jsonData);

    public void trackAdjustEventStart(String eventToken);

    public void trackAdjustEventGreeting(String eventToken);

    public void trackAdjustEventAccess(String eventToken);

    public void trackAdjustEventUpdated(String eventToken);

    public String getDeviceID();

    public String getDeviceInfoForLighthouse();

    public int getSystemVersionCode();

    public int getClientVersionCode();

    public String getPackageName();

    public String getAppName();

    public String getChannel();

    public String getBrand();

    public void saveGameUrl(String gameUrl);

    public void saveAccountInfo(String plainText);

    public String getAccountInfo();

    public String getAdjustDeviceID();

    public String getGoogleADID();

    public String getIDFA();

    public String getReferID();

    public String getAgentID();

    public void setCocosData(String key, String value);

    public String getCocosData(String key);

    public String getCocosAllData();

    public String getLighterHost();

    public int getBridgeVersion();

    public boolean isFacebookEnable();

    public String getTDTargetCountry();

    public void openUrlByBrowser(String url);

    public void openUrlByWebView(String url);

    public void openApp(String target, String fallbackUrl);

    public void loadUrl(String url);

    public void goBack();

    public void close();

    public void refresh();

    public void clearCache();

    public void saveImage(String url);

    public void savePromotionMaterial(String materialUrl);

    public void synthesizePromotionImage(String qrCodeUrl, int size, int x, int y);

    public void shareUrl(String url);

    public void loginFacebook();

    public void logoutFacebook();

    public void preloadPromotionImage(String imageUrl);

    public void shareToWhatsApp(String text);

    public boolean isHttpDnsEnable();

    public String httpdns(String host);

    public void httpdnsInit(String hosts);

    public String httpdnsRequestSync(String req, byte[] body);

    public void httpdnsRequestAsync(String req, byte[] body);

    public void httpdnsWsOpen(String req);

    public String httpdnsWsSend(String req, byte[] body);

    public void httpdnsWsClose(String req);

    public String httpdnsWsConnected(String req);

    public String getBuildVersion();

    public void onAnalysisStart(String accid, long cretime);

    public void onAnalysisEnd();

    public String memoryInfo();

    public boolean isEmulator();

    public String commonData();

    public void exitApp();

    public void handleNotification();
}

File: app-sdk/src/main/java/book/sdk/bridge/BridgeInterface.kt
package book.sdk.bridge

interface BridgeInterface {
    fun nativeLog(tag: String?, msg: String?)
    fun copyText(text: String?)
    fun getCopiedText(): String?
    fun showNativeToast(toast: String?)
    fun initAdjustID(adjustAppID: String?)
    fun trackAdjustEvent(eventToken: String?, jsonData: String?)
    fun trackAdjustEventStart(eventToken: String?)
    fun trackAdjustEventGreeting(eventToken: String?)
    fun trackAdjustEventAccess(eventToken: String?)
    fun trackAdjustEventUpdated(eventToken: String?)
    fun getDeviceID(): String?
    fun getDeviceInfoForLighthouse(): String?
    fun getSystemVersionCode(): Int
    fun getClientVersionCode(): Int
    fun getPackageName(): String?
    fun getAppName(): String?
    fun getChannel(): String?
    fun getBrand(): String?
    fun saveGameUrl(gameUrl: String?)
    fun saveAccountInfo(plainText: String?)
    fun getAccountInfo(): String?
    fun getAdjustDeviceID(): String?
    fun getGoogleADID(): String?
    fun getIDFA(): String?
    fun getReferID(): String?
    fun getAgentID(): String?
    fun setCocosData(key: String?, value: String?)
    fun getCocosData(key: String?): String?
    fun getCocosAllData(): String?
    fun getLighterHost(): String?
    fun getBridgeVersion(): Int
    fun isFacebookEnable(): Boolean
    fun getTDTargetCountry(): String?
    fun openUrlByBrowser(url: String?)
    fun openUrlByWebView(url: String?)
    fun openApp(target: String?, fallbackUrl: String?)
    fun loadUrl(url: String?)
    fun goBack()
    fun close()
    fun refresh()
    fun clearCache()
    fun saveImage(url: String?)
    fun savePromotionMaterial(materialUrl: String?)
    fun synthesizePromotionImage(qrCodeUrl: String?, size: Int, x: Int, y: Int)
    fun shareUrl(url: String?)
    fun loginFacebook()
    fun logoutFacebook()
    fun preloadPromotionImage(imageUrl: String?)
    fun shareToWhatsApp(text: String?)
    fun isHttpDnsEnable(): Boolean
    fun httpdns(host: String?): String?
    fun httpdnsInit(hosts: String?)
    fun httpdnsRequestSync(req: String?, body: ByteArray?): String?
    fun httpdnsRequestAsync(req: String?, body: ByteArray?)
    fun httpdnsWsOpen(req: String?)
    fun httpdnsWsSend(req: String?, body: ByteArray?): String?
    fun httpdnsWsClose(req: String?)
    fun httpdnsWsConnected(req: String?): String?
    fun getBuildVersion(): String?
    fun onAnalysisStart(accid: String?, cretime: Long)
    fun onAnalysisEnd()
    fun memoryInfo(): String?
    fun isEmulator(): Boolean
    fun commonData(): String?
    fun exitApp()
    fun handleNotification()
    fun onWebViewLoadChanged(json: String?)
}
